### PR TITLE
fix matplotlib imports, pd.util

### PR DESCRIPTION
**Pull request by Dr-Irv**
CI was failing due to matplotlib changes.  This fixes those issues.

### CODE REVIEW

File: pandas-stubs/plotting/_core.pyi [hunk 1]
```diff
@@ -239,7 +239,7 @@ class PlotAccessor:
         **kwargs: Any,
     ) -> pd.Series: ...
     @overload
-    def line(
+    def line(  # type: ignore[misc]
```
Comment (twoertwein) [inline pandas-stubs/plotting/_core.pyi:242]: Just for me to understand why these changes are needed: mypy&pyright cannot resolve Axis, so it is treated as Any/Unknown, therefore the overloads overlap. Is that correct?

File: tests/test_pandas.py [hunk 2]
```diff
@@ -13,6 +13,7 @@
 import pandas as pd
 from pandas import Grouper
 from pandas.api.extensions import ExtensionArray
+import pandas.util as pdutil
```
Comment (twoertwein) [inline tests/test_pandas.py:16]: Maybe add a `# TODO: github.com/pandas-dev/pandas/issues/55023` I assume this would fix the workaround?

File: tests/test_plotting.py [hunk 3]
```diff
@@ -1,6 +1,8 @@
 import io
 from typing import Any
 
+from matplotlib.axes import Axes
+from matplotlib.figure import Figure
```
Comment (twoertwein) [inline tests/test_plotting.py:5]: I assume this just hides the explicit re-export issue (I'm surprised that mypy/pyright are fooled that easily).

**Review by twoertwein — APPROVED** [review]
Except for adding a comment about `pd.util` it looks good to me.

(Should probably open an issue/pr at matplotlib so that Axis/Figure are ex-exported)

File: pandas-stubs/plotting/_core.pyi (hunk 1 above)
Comment (Dr-Irv) [reply pandas-stubs/plotting/_core.pyi:242]: I think this is because `matplotlib` changed the way that `Axes` is declared.

File: tests/test_plotting.py (hunk 3 above)
Comment (Dr-Irv) [reply tests/test_plotting.py:5]: No, I think `matplotlib` got updated, and now `plt.Axes` and `plt.Figure` aren't valid any more in code from a type checking perspective.

File: tests/test_pandas.py (hunk 2 above)
Comment (Dr-Irv) [reply tests/test_pandas.py:16]: I think so.  In next commit.

File: pandas-stubs/plotting/_core.pyi (hunk 1 above)
Comment (twoertwein) [reply pandas-stubs/plotting/_core.pyi:242]: I looked at their code: maybe the `from ... import *` or the many class decorators throw mypy/pyright off.

**Comment by twoertwein** [comment]
Thanks @Dr-Irv !

**Comment by Dr-Irv** [comment]
Create the matplotlib issue at https://github.com/matplotlib/matplotlib/issues/26812

File: pandas-stubs/plotting/_core.pyi (hunk 1 above)
Comment (ksunden) [reply pandas-stubs/plotting/_core.pyi:242]: I don't think this has to do with `matplotlib`, but is rather a change in `mypy` (Saw similar things in `xarray`)

Here, I think the problem may actually be that you have a default value indicated for both the `True` and `False` case of the subplots kwarg, when I think that should only apply in one case.

Comment (twoertwein) [reply pandas-stubs/plotting/_core.pyi:242]: @ksunden I think you are right - one of the overloads should require a user-provided input!

Comment (Dr-Irv) [reply pandas-stubs/plotting/_core.pyi:242]: > @ksunden I think you are right - one of the overloads should require a user-provided input!

Created #777
